osu.Game.Rulesets.Osu/Skinning/Default/SliderBall.cs

```diff
@@ -103,14 +103,22 @@ private set
                 tracking = value;
 
                 if (InputTracksVisualSize)
-                    followCircle.ScaleTo(tracking ? 2.4f : 1f, 300, Easing.OutQuint);
+                {
+                    if (tracking)
+                        followCircle.ScaleTo(2f, 200, Easing.OutQuint);
+                    else
+                        followCircle.ScaleTo(1.6f, 200, Easing.None);
+                }
                 else
                 {
                     // We need to always be tracking the final size, at both endpoints. For now, this is achieved by removing the scale duration.
-                    followCircle.ScaleTo(tracking ? 2.4f : 1f);
+                    followCircle.ScaleTo(tracking ? 2f : 1f);
                 }
 
-                followCircle.FadeTo(tracking ? 1f : 0, 300, Easing.OutQuint);
```

Comment:
you're changing the defaults. i think you intended to only change legacy skins.

Reply:
I may be misunderstanding, but every single `DrawableSlider` instantiates this specific `SliderBall` class. So I'm not sure why `SliderBall` is in Defaults anyway since it's always used, and I don't see a way to only affect legacy skins.

https://github.com/ppy/osu/blob/512af8809ed9a94f98338593472b1ead628e9212/osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSlider.cs#L32
https://github.com/ppy/osu/blob/512af8809ed9a94f98338593472b1ead628e9212/osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSlider.cs#L76-L82

And the legacy slider ball class is super empty:
https://github.com/ppy/osu/blob/512af8809ed9a94f98338593472b1ead628e9212/osu.Game.Rulesets.Osu/Skinning/Legacy/LegacySliderBall.cs

Shouldn't all this `SliderBall` functionality be inside a `DrawableSliderBall` class (i.e. like `DrawableSliderHead`)?

Reply:
You'll have to refactor to allow this. There's plenty of examples.